File: solutions/python/atbash-cipher/1/atbash_cipher.py
plain = list('abcdefghijklmnopqrstuvwxyz')
cipher = list('zyxwvutsrqponmlkjihgfedcba')

def encode(plain_text):
    ciphered_text = ""
    for letter in plain_text.lower():
        if letter in ",. ":
            continue
        if letter.isdigit():
            ciphered_text += letter
        else:
            ciphered_text += cipher[plain.index(letter)]
        if (len(ciphered_text) - ciphered_text.count(" ")) % 5 == 0:
            ciphered_text += " "
    return ciphered_text.strip()

def decode(ciphered_text):
    plain_text = ""
    for letter in ciphered_text:
        if letter == " ":
            continue
        if letter.isdigit():
            plain_text += letter
        else:
            plain_text += plain[cipher.index(letter)]
    return plain_text

File: solutions/cpp/atbash-cipher/2/atbash_cipher.cpp
#include "atbash_cipher.h"
#include <bits/stdc++.h>
using namespace std;
namespace atbash_cipher {

// TODO: add your solution here
    string plain =  "abcdefghijklmnopqrstuvwxyz";
    string cipher = "zyxwvutsrqponmlkjihgfedcba";
    string encode(string str){
        string result{};
        int count = 0;
        for(char c : str){
            if(c == 32)continue;
            if(tolower(c) > 96) result += cipher.at(plain.find(tolower(c)));
            else if(c == 44 or c == 46) continue;
            else result += c;
            count++;
            if (count == 5 and str.find(c) != str.length() - 2) {
                result += " ";
                count = 0;
            }
        }
        return result;
    }

    string decode(string str){
        string result{};
        for(char c : str){
            if(c == 32)continue;
            if(tolower(c) > 96) {
                result += plain.at(cipher.find(tolower(c)));
                continue;
            }
            result += c;
        }
        return result;
    }

}  // namespace atbash_cipher
